src/pages/View/View.js

```diff
@@ -11,50 +11,30 @@ const DAY_END = '20:00';
 
 const colors = ['red', 'orange', 'yellow', 'green', 'blue', 'purple'];
 
+// Moment object with KidsAbility specific date format
+const mnt = (date) => moment(date, 'DD-MMM-YY')
+
 // compare function to sort time slots
 const compareFunction = (a, b) => {
-    if (moment(a.Date).hour(a.Start.split(":")[0]).minute(a.Start.split(":")[1]) < 
-        moment(b.Date).hour(b.Start.split(":")[0]).minute(b.Start.split(":")[1])) {
-        return -1;
-    }
-    
-    if (moment(a.Date).hour(a.Start.split(":")[0]).minute(a.Start.split(":")[1]) > 
-        moment(b.Date).hour(b.Start.split(":")[0]).minute(b.Start.split(":")[1])) {
-        return 1;
-    }
-    
-    if (a.Start < b.Start) return 1;
-    if (b.Start < a.Start) return -1;
-    
-    return 0;
-}
-
-//compare function for time "h:mm"
-const compareTime = (a, b) => {
-    let hourA = parseInt(a.split(":")[0])
-    let hourB = parseInt(b.split(":")[0]);
-    let minA = parseInt(a.split(":")[1])
-    let minB = parseInt(b.split(":")[1])
-    if (hourA < hourB){
-        return 1;
-    }
-    if (hourA > hourB) {
+    const aTime = mnt(a.Date)
+        .hour(a.Start.split(":")[0])
+        .minute(a.Start.split(":")[1])
+    const bTime = mnt(b.Date)
+        .hour(b.Start.split(":")[0])
+        .minute(b.Start.split(":")[1])
+    // TODO: isAfter, isBefore?
+    if (aTime < bTime) {
         return -1;
-    } 
-    if (minA < minB) {
+    } else if (aTime > bTime) {
         return 1;
+    } else {
+      return 0;
     }
-    if (minA > minB) {
-        return -1;
-    }
-    return 0;
 }
 
 // create an array of available times grouped by weeks
 const processData = (data) => {
-    const sortedData = data
-        .flatMap(x => x)
```

Comment:
`Array.prototype.flatMap` is not implemented in Firefox. Also, I'm pretty sure `flatMap` is not needed in this function.

Reply:
I don't think it is needed anymore - since we have fixed the unnecessary schema nesting.

I didn't review code that critically the last term - if I worked I didn't ask many questions.